### PR TITLE
implement hotkey functionality

### PR DESCRIPTION
**Pull request by zulqarnainhanif**
## Summary

implement ctrl/cmd+s functionality
implemented the functionality for grouping and duplicating nodes from hierarchy as well.

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] Changes have been manually QA'd 
- [ ] Changes reviewed by at least 2 approved reviewers

## References

https://github.com/XRFoundation/XREngine/issues/4113

## Reviewers

@HexaField

### CODE REVIEW

**Comment by speigg** [comment]
Awesome

File: packages/editor/src/functions/utils.ts [hunk 1]
```diff
@@ -34,8 +34,23 @@ export function unique(arr, maybeComp) {
...
+
+  const iOS13_iPad = navigator.platform === 'MacIntel'
+
+  const iOS1to12quirk = () => {
+    var audio = new Audio() // temporary Audio object
+    audio.volume = 0.5 // has no effect on iOS <= 12
+    return audio.volume === 1
+  }
+
+  return iOS_1to12 || iOS13_iPad || iOS1to12quirk()
```
Comment (speigg) [inline packages/editor/src/functions/utils.ts:49]: Can you explain why this is necessary?

**Review by speigg — APPROVED** [review]

File: packages/editor/src/functions/utils.ts (hunk 1 above)
Comment (zulqarnainhanif) [reply packages/editor/src/functions/utils.ts:49]: I have the latest mac and it kept showing ctrl + s for it while it should show cmd + s.

The reason behind this is that for latest macs it return `MacIntel` as navigator.platform which wasn't being caught in the previous solution.

So i found an interesting solution on stackoverflow for this which i implemented above.

Comment (speigg) [reply packages/editor/src/functions/utils.ts:49]: Does this solution also work with older Macs?

**Review by HexaField — APPROVED** [review]
Group and duplication functionality seems to not be working for me, but I don't mind as we are currently refactoring the way the editor nodes work, so we can revisit the specific funcitonality at a later time. The keybindings work, so I'm happy with this!

**Pull request merged**